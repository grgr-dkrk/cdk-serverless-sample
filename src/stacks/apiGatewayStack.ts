import { Construct, Stack, StackProps } from '@aws-cdk/core'
import { Function as LambdaFunction } from '@aws-cdk/aws-lambda'
import { LambdaIntegration, RestApi } from '@aws-cdk/aws-apigateway'
import { DeployEnv } from '../config'

type Props = StackProps & {
  lambda: LambdaFunction
  resourceName: string
  deployEnv: DeployEnv
  apiName?: string
  method?: string
}

/**
 * Stack for aws-api-gateway
 */
export class APIGatewayStack extends Stack {
  public readonly restAPI: RestApi
  constructor(scope: Construct, id: string, props: Props) {
    super(scope, id, props)

    const { apiName, lambda, method, resourceName } = props

    // rest api
    this.restAPI = new RestApi(this, `ApiGateway${id}`, {
      restApiName: apiName || 'myApi',
    })

    // add Resource
    const resource = this.restAPI.root.addResource(resourceName)

    // Integration
    const lambdaIntegration = new LambdaIntegration(lambda)

    // add Method
    resource.addMethod(method || 'GET', lambdaIntegration)
  }
}
