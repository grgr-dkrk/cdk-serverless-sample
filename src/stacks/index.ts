export { LambdaStack } from './lambdaStack'
export { APIGatewayStack } from './APIGatewayStack'
