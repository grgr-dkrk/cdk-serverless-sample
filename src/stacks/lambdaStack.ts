import { Construct, Duration, Stack, StackProps } from '@aws-cdk/core'
import { Code, Function as LambdaFunction, Runtime } from '@aws-cdk/aws-lambda'
import { DeployEnv } from '../config'

type Props = StackProps & {
  code: Code
  handlerName: string
  deployEnv: DeployEnv
  functionName?: string
}

/**
 * Stack for aws-lambda
 */
export class LambdaStack extends Stack {
  public readonly lambda: LambdaFunction
  constructor(scope: Construct, id: string, props: Props) {
    super(scope, id, props)

    const { code, handlerName, functionName } = props

    this.lambda = new LambdaFunction(this, `LambdaFunc${id}`, {
      code,
      functionName: functionName || 'Lambda',
      handler: `index.${handlerName}`,
      runtime: Runtime.NODEJS_10_X,
      timeout: Duration.seconds(10),
    })
  }
}
