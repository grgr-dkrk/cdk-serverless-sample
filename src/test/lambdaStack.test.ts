import { Code } from '@aws-cdk/aws-lambda'
import * as cdk from '@aws-cdk/core'
import { LambdaStack } from '../stacks'
import * as path from 'path'
import { SynthUtils } from '@aws-cdk/assert'
import { envMock } from './__mocks__/env'

describe('lambdaStack', () => {
  const app = new cdk.App()
  const lambdaStack = new LambdaStack(app, 'TestLambdaStack', {
    code: Code.fromAsset(path.resolve(__dirname, './__mocks__/functions')),
    deployEnv: 'staging',
    handlerName: 'LambdaFunctionMock',
    env: envMock,
  })
  it('snapshot', () => {
    expect(SynthUtils.toCloudFormation(lambdaStack)).toMatchSnapshot(
      'lambdaTest',
    )
  })
})
