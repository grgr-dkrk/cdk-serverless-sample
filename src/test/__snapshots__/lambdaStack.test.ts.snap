// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lambdaStack snapshot: lambdaTest 1`] = `
Object {
  "Parameters": Object {
    "AssetParameterse99417ba66563098507650fbd4432a9f20f8dd9a9dced4093830a674d642af3aArtifactHash3A07462E": Object {
      "Description": "Artifact hash for asset \\"e99417ba66563098507650fbd4432a9f20f8dd9a9dced4093830a674d642af3a\\"",
      "Type": "String",
    },
    "AssetParameterse99417ba66563098507650fbd4432a9f20f8dd9a9dced4093830a674d642af3aS3Bucket1B447F69": Object {
      "Description": "S3 bucket for asset \\"e99417ba66563098507650fbd4432a9f20f8dd9a9dced4093830a674d642af3a\\"",
      "Type": "String",
    },
    "AssetParameterse99417ba66563098507650fbd4432a9f20f8dd9a9dced4093830a674d642af3aS3VersionKey2DB78F77": Object {
      "Description": "S3 key for asset version \\"e99417ba66563098507650fbd4432a9f20f8dd9a9dced4093830a674d642af3a\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "LambdaFuncTestLambdaStack2EE31B0B": Object {
      "DependsOn": Array [
        "LambdaFuncTestLambdaStackServiceRole4EBB1EA0",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameterse99417ba66563098507650fbd4432a9f20f8dd9a9dced4093830a674d642af3aS3Bucket1B447F69",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameterse99417ba66563098507650fbd4432a9f20f8dd9a9dced4093830a674d642af3aS3VersionKey2DB78F77",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameterse99417ba66563098507650fbd4432a9f20f8dd9a9dced4093830a674d642af3aS3VersionKey2DB78F77",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "FunctionName": "Lambda",
        "Handler": "index.LambdaFunctionMock",
        "Role": Object {
          "Fn::GetAtt": Array [
            "LambdaFuncTestLambdaStackServiceRole4EBB1EA0",
            "Arn",
          ],
        },
        "Runtime": "nodejs10.x",
        "Timeout": 10,
      },
      "Type": "AWS::Lambda::Function",
    },
    "LambdaFuncTestLambdaStackServiceRole4EBB1EA0": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
